cpp/src/parquet/file_reader.cc

```diff
@@ -66,7 +66,8 @@ static constexpr int64_t kMaxDictHeaderSize = 100;
 RowGroupReader::RowGroupReader(std::unique_ptr<Contents> contents)
     : contents_(std::move(contents)) {}
 
-std::shared_ptr<ColumnReader> RowGroupReader::Column(int i) {
+std::shared_ptr<ColumnReader> RowGroupReader::Column(
+    int i, std::optional<ReaderProperties> prop) {
```

Comment:
This does not look better than the proposed `ColumnProperties` from @mapleFU

Reply:
Seems that `RowGroupReader` would already has `properties_`. Could you use that?

Reply:
@jp0317 In `WriterProperties`, it has a Column-specific writer property, and when column-level property doesn't exist, it will refer to original property. Why not just use that to handling?

And under that, `Column` will get `buffer_size()` without breaking the previous interface.

Reply:
Is it an overkill to setup a column-level buffer size of each row group? If you really need control in this granularity, you can create one reader for each row group. Probably enabling buffered stream in this case is not a good idea as its behavior is not that predicable.

ReaderProperties is immutable once reader has been created, mutating it seems not a good idea to me.